backend/services/core/migrations/20230822094308_add_custom_author_type_to_articles.ts

```diff
@@ -0,0 +1,106 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+  // Add custom authors table
+  await knex.schema.createTable('custom_authors', (table) => {
+    table.uuid('id').primary().defaultTo(knex.raw('gen_random_uuid()'));
+    table.string('name').notNullable().comment('Swedish name of custom author');
+    table.string('name_en').comment('English name of custom author, possibly null');
+    table.string('image_url').comment('Image url of custom author, possibly null');
+    table.timestamps(true, true);
+  });
+  await knex.schema.createTable('custom_author_roles', (table) => {
+    table.uuid('id').primary().defaultTo(knex.raw('gen_random_uuid()'));
+    table.uuid('custom_author_id').unsigned().notNullable().references('custom_authors.id')
+      .onDelete('CASCADE');
+    table.string('role').unsigned().notNullable().comment('should use role syntax, such as dsek.styr or dsek.stab.noll');
+    table.timestamps(true, true);
+  });
+  // Add new "authors" table
+  await knex.schema.createTable('authors', (table) => {
+    table.uuid('id').primary().defaultTo(knex.raw('gen_random_uuid()'));
+    table.uuid('member_id').notNullable().references('members.id').onDelete('CASCADE');
+    table.uuid('mandate_id').unsigned().nullable().references('mandates.id')
+      .onDelete('SET NULL');
+    table.uuid('custom_id').unsigned().nullable().references('custom_authors.id')
+      .onDelete('SET NULL');
+    table.timestamps(true, true);
+    table.enu('type', ['Member', 'Mandate', 'Custom'], { useNative: true, enumName: 'author_type' }).notNullable().defaultTo('Member');
+    table.uuid('temp_article_id').unsigned().notNullable().references('articles.id')
+      .comment('Temporary column during migration');
+  });
+  // Get all articles posted by a member, not as a mandate
+  const memberAuthorArticles = await knex('articles').select('id', 'author_id', 'author_type', 'published_datetime', 'latest_edit_datetime')
+    .where({ author_type: 'Member' });
+  // Get all articles posted as a mandate
+  const mandateAuthorArticles = await knex('articles').select('articles.id', 'author_id', 'author_type', 'mandates.member_id', 'mandates.id as mandate_id', 'published_datetime', 'latest_edit_datetime')
+    .join('mandates', 'mandates.id', '=', 'articles.author_id')
+    .where({ author_type: 'Mandate' });
+  // Create rows in authors table for each article posted as a member
+  if (memberAuthorArticles.length > 0) {
+    await knex('authors')
```

Comment:
This is more of an issue with naming than anything else. I think I was getting authors and custom_authors mixed up when I wrote this comment. Either way that confusion is in line with my review summary.

Reply:
As mentioned in the original comment of the PR, I've opted for having one author row as otherwise updating or modifying author could lead to problems. The reason that authors is its' own table to begin with is JUST so the format could be used for more than just news, for example events or booking requests.

It's not for data de-duplication but for schema consistency (two words which I don't think are real but I'm sure you understand what I mean).

Reply:
It's really just a relationtable between articles, members,mandates and CustomAuthors.